internal/host/transition_test.go

```diff
@@ -5988,3 +5989,153 @@ func validateEqualProgress(p1, p2 *models.HostProgressInfo) {
 		Expect(time.Time(p1.StageUpdatedAt).Equal(time.Time(p2.StageUpdatedAt))).To(BeTrue())
 	}
 }
+
+type testState struct {
+	state string
+}
+
+func newTestState(state string) *testState {
+	return &testState{
+		state: state,
+	}
+}
+
+func (ts *testState) State() stateswitch.State {
+	return stateswitch.State(ts.state)
+}
+
+func (ts *testState) SetState(state stateswitch.State) error {
+	ts.state = string(state)
+	return nil
+}
+
+var _ = Describe("State machine test", func() {
+	var (
+		ctx                   = context.Background()
+		mockController        *gomock.Controller
+		mockTransitionHandler *MockTransitionHandler
+		stateMachine          stateswitch.StateMachine
+		refreshHostArgs       TransitionArgsRefreshHost
+		testState             *testState
+	)
+
+	BeforeEach(func() {
+		mockController = gomock.NewController(GinkgoT())
+
+		mockTransitionHandler = NewMockTransitionHandler(mockController)
+
+		mockTransitionHandler.EXPECT().HasStatusTimedOut(gomock.Any()).Return(func(_ stateswitch.StateSwitch, _ stateswitch.TransitionArgs) (bool, error) {
+			return false, nil
+		}).AnyTimes()
+
+		mockTransitionHandler.EXPECT().PostRefreshHost(gomock.Any()).Return(
```

Comment:
IMO using any times will make the tests more shallow

Reply:
My attitude towards mocks call counts are -

- If I use a mock to test that a particular external function is actually being called as expected, I probably want to specify the exact number of times I expect it to be called. 

- If I use mock just so my code can run without issues, I use `AnyTimes()`, because I don't want to be too opinionated regarding how many times it gets called - it might change in the future, and my test couldn't care less. Since my test is not dealing with testing that this particular call happens, my test also doesn't care how many times it gets called.

Reply:
I agree with Omer on that one. From my experience, when writing a big test I tend to spend more time on figuring out _what_ and _how many times_ I need to mock to get it passing instead of focusing on whether my feature is being tested properly. I think AnyTimes, Maybe & co. are perfectly acceptable as long as they are not core of the test

Reply:
then maybe it show that the code need to be more simple and break it more. 
not a deal breaker for me.